src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.WebApi.Logging
+{
+    /// <summary>
+    /// Request tracing middleware component to log every incoming HTTP request.
+    /// </summary>
+    public class RequestTrackingMiddleware
```

Comment:
I'm missing the configuration aspect a bit from the issue:

> Configuration should be provided in what should be logged:
> - Headers only, body only, headers and body, etc.

Can we provide an options/configuration approach similar to other flow?

Reply:
No those are present, you can override the middleware like the issue described was the way to go.

Reply:
Inheritance should only be required if you want to sanitize, not for those things as per the issue:

> Things to keep in mind:
> 
> - Configuration should be provided in what should be logged:
>    - Headers only, body only, headers and body, etc.
> - We should provide sanitization out-of-the-box, with possibility to provide custom sanitization *(based on inheritance)*

Reply:
As discussed, people will be able to:
- Sanitize headers by providing a list of sensitive headers to remove; and/or by using inheritance
- What needs to be logged will be made configurable (headers only, body only, none which means just URL)

Reply:
Done that.